CONTRIBUTING.md

```diff
@@ -391,3 +392,53 @@ Here is an example of such lightweight workflow in Linux environment (Debian-bas
 Maintainer: [@stefanseefeld](https://github.com/stefanseefeld)
 
 _TODO_
+
+## Guidelines
+
+Boost.GIL is a more than a decade old mature library maintained by several
+developers with help from a couple of dozens contributors.
+It is important to maintain consistent design, look and feel.
+Thus, below a few basic guidelines are listed.
+
+First and foremost, make sure you are familiar with the official
+[Boost Library Requirements and Guidelines](https://www.boost.org/development/requirements.html).
+
+Second, strive for writing idiomatic C++11, clean and elegant code.
+
+**NOTE:** *The Boost.GIL source code does not necessary represent clean and elegant
+code to look up to. The library has recently entered the transition to C++11.
+Major refactoring overhaul is ongoing.*
+
+Maintain structure your source code files according to the following guidelines:
+
+* Name files in meaningful way.
+* Put copyright and license information in every file
+* If your changes [meet a certain threshold of originality](https://www.boost.org/users/license.html),
+  add yourself to the copyright notice. Do not put any additional authorship or
+  file comments (eg. no `\file` for Doxygen), revision information, etc.
+* In header, put `#include` guard based on header path and file name
+    ```
+    #ifndef BOOST_GIL_<DIR1>_<DIR2>_<FILE>_HPP
+    #define BOOST_GIL_<DIR1>_<DIR2>_<FILE>_HPP
+    ...
+    #endif
+    ```
+* Group `#include` directives and order alphabetically within each group, as follows:
+    * In headers of GIL core and extensions:
```

Comment:
I think this is way too much detail. We are not IBM or Microsoft (or any of those big corporations). While I appreciate consistency, I wouldn't want to establish policies dictating the exact ordering of includes.

Reply:
Order of includes *from specific to general*, quite helps to ensure each header includes all that is required. In fact, this already helped to identify a couple of places where includes where missing:

Reply:
I believe this has been resolved too. Do you see any more changes necessary @stefanseefeld ?

Reply:
It all looks good to me now. Thanks for working on it !

Reply:
Right, I was hoping to get your Approve hit :) Thx